extensions/amp-form/0.1/form-validators.js

```diff
@@ -345,9 +370,17 @@ export class AbstractCustomValidator extends FormValidator {
     }
     input[VISIBLE_VALIDATION_CACHE] = validation;
 
-    this.resources.mutateElement(input, () =>
-      input.setAttribute('aria-invalid', 'true')
-    );
+    let validationId = validation.getAttribute('id');
+    if (!validationId) {
+      validationId = this.createUniqueAriaDescId_();
+      validation.setAttribute('id', validationId);
+    }
+
+    this.resources.mutateElement(input, () => {
+      input.setAttribute('aria-invalid', 'true');
```

Comment:
Do the aria attributes force layout? I think you might be able to do without `mutateElement` here and below.

Reply:
I see now that there are other `mutateElement` calls in this file that wrap aria attribute reads/writes, like some lines that I had written 😭. We should probably clean that up to avoid confusing people. There is a minor performance benefit from not wrapping these unnecessarily.